src/account/accountsReducer.js

```diff
@@ -110,6 +112,34 @@ export default (state: AccountsState = initialState, action: Action): AccountsSt
     case ACCOUNT_REMOVE:
       return accountRemove(state, action);
 
+    case EVENT: {
+      const { event } = action;
+      switch (event.type) {
+        case 'restart': {
+          const { zulip_feature_level, zulip_version } = event;
+          if (zulip_feature_level !== undefined && zulip_version !== undefined) {
+            // TODO (?): Detect if these are different from the values
+            // we had before, so we know it's an upgrade, not just a
+            // restart. Then, implement logic like "be sure to refetch
+            // from scratch within N hours" (to avoid thundering
+            // herding the server).
+            return [
+              {
+                ...state[0],
+                zulipVersion: new ZulipVersion(zulip_version),
+                zulipFeatureLevel: zulip_feature_level ?? 0,
```

Comment:
Do we expect `zulip_feature_level` to be null ever? It seems weird be me tat we need the `?? 0` here given the `undefined` check above.

Reply:
Good point; no, we don't expect any servers to send `null` for this (I confirmed by checking `ApiResponseServerSettings`).

Reply:
In that case, could you remove the `?? 0`? (perhaps you'd need to change the check from `zulip_feature_level !== undefined` to `zulip_feature_level != null` or similar in order to make flow happy, I'm not sure).

Reply:
> In that case, could you remove the `?? 0`

Oops—I thought I'd done that for this revision; looks like not. 🙂 Will fix.